### ISSUE TITLE
`linkerd-init` and `linkerd-proxy` sidecars violate PodSecurity "restricted:v1.24"

### ISSUE THREAD

**Issue by brsolomon-deloitte** [linkerd/linkerd2#10232]
### What is the issue?

Linkerd when installed via Helm chart is not able to run its sidecars when using a `Restricted` [Pod Security Standard](https://kubernetes.io/docs/concepts/security/pod-security-standards/).

This leads to the following error when attempting a `kubectl rollout`:

> (combined from similar events): Error creating: pods "gatekeeper-audit-d4f56788f-9xqz9" is forbidden: violates PodSecurity "restricted:v1.24": unrestricted capabilities (container "linkerd-init" must not include "NET_ADMIN", "NET_RAW" in securityContext.capabilities.add), runAsNonRoot != true (pod or container "linkerd-proxy" must set securityContext.runAsNonRoot=true), seccompProfile (pod or containers "linkerd-init", "linkerd-proxy" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")

### How can it be reproduced?

I observed this when meshing [gatekeeper](https://github.com/open-policy-agent/gatekeeper) on Amazon EKS 1.24. First annotate the namespace, then restart the deployments.

```
kubectl annotate ns gatekeeper-system 'linkerd.io/inject=enabled'
kubectl rollout restart -n gatekeeper-system deployment -l 'app=gatekeeper'
```

### Logs, error output, etc

> (combined from similar events): Error creating: pods "gatekeeper-audit-d4f56788f-9xqz9" is forbidden: violates PodSecurity "restricted:v1.24": unrestricted capabilities (container "linkerd-init" must not include "NET_ADMIN", "NET_RAW" in securityContext.capabilities.add), runAsNonRoot != true (pod or container "linkerd-proxy" must set securityContext.runAsNonRoot=true), seccompProfile (pod or containers "linkerd-init", "linkerd-proxy" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")

### output of `linkerd check -o short`

```
$ linkerd check -o short
Status check results are √
```

### Environment

Amazon EKS 1.24

### Possible solution

The Helm chart for `linkerd-control-plane` should allow some configurability for the `securityContext` that is used by `linkerd-init` and `linkerd-proxy` sidecars.

### Would you like to work on fixing this bug?

no

**Comment by brsolomon-deloitte** [comment]
Config could be added here: 
https://github.com/linkerd/linkerd2/blob/f4698538af7b05bbf89ff7c63eb095957e3d0518/charts/linkerd-control-plane/values.yaml#L231

**Comment by alpeb** [comment]
As of `edge-23.1.1` we're including the annotation `pod-security.kubernetes.io/enforce: privileged` in the linkerd namespace to follow the Pod Security Standard with the `privileged` policy (note though that extensions had an issue with this, that got fixed in today's `edge-23.1.2`). We also added more defaults to all the security contexts. Note that those are the minimal privileges linkerd is able to run on. If you need lower privileges, then you need to run linkerd in [CNI mode](https://linkerd.io/2.12/features/cni/), where all the workloads run under the `restricted` policy.

**Comment by brsolomon-deloitte** [comment]
not stale
